hub/core/chunk_engine.py

```diff
@@ -239,17 +239,21 @@ def get_chunk_commit(self, chunk_name) -> str:
         cur_node: CommitNode = self.version_state["commit_node"]
         while cur_node is not None:
             commit_id = cur_node.commit_id
-            chunk_list_key = get_tensor_commit_chunk_list_key(self.key, commit_id)
+            chunk_set_key = get_tensor_commit_chunk_set_key(self.key, commit_id)
             try:
-                chunk_list = self.meta_cache.get_cachable(
-                    chunk_list_key, CommitChunkList
-                ).chunks
+                # the first commit doesn't contain a chunk set, don't repeatedly try to fetch from storage
```

Comment:
this is the major fix. it was repeatedly trying to get chunk set for first commit, which never exists, leading to a lot of repeated s3 calls with s3 get errors that slowed us down.

Reply:
the rest of the PR is essentially replacing ChunkList->ChunkSet, which will lead to improvements with bigger datasets